packages/redirect/src/class-redirect.php

```diff
@@ -55,8 +59,18 @@ public static function get_url( $source, $args = array() ) {
 		$args          = wp_parse_args( $args, array( 'site' => self::build_raw_urls( get_home_url() ) ) );
 		$accepted_args = array( 'site', 'path', 'query', 'anchor' );
 
+		$source_key = 'source';
+
+		if ( 0 === strpos( $source, 'https://' ) ) {
+			$source_key = 'url';
+			$source_url = \wp_parse_url( $source );
+
+			// discard any query and fragments.
+			$source = $source_url['scheme'] . '://' . $source_url['host'] . ( isset( $source_url['path'] ) ? $source_url['path'] : '' );
```

Comment:
Do we need to dynamically add `$source_url['scheme'` here, given that we'll only support `https` links as per above?

```suggestion
			$source = 'https://' . $source_url['host'] . ( isset( $source_url['path'] ) ? $source_url['path'] : '' );
```